components/EmotionList.js

```diff
@@ -1,30 +1,72 @@
 import styled from "styled-components";
 import Link from "next/link";
 import EmotionCard from "./EmotionCard";
+import { useState } from "react";
+import { emotions } from "@/lib/emotions";
 
 export default function EmotionList({ objects }) {
+  const [selectedFilter, setSelectedFilter] = useState("");
+
+  const filteredObjects = selectedFilter
+    ? objects.filter((object) => object.emotion === selectedFilter)
+    : objects;
+
   return (
-    <StyledEmotionList>
-      {objects.length === 0 ? (
+    <>
+      <StyledHeadline>Filter emotion type</StyledHeadline>
+      <ul>
+        {emotions.map((emotion) => (
+          <StyledFilterList key={emotion}>
+            <StyledFilterButtons
+              onClick={() =>
+                setSelectedFilter(emotion === selectedFilter ? "" : emotion)
+              }
+              $isSelected={emotion === selectedFilter}
+            >
+              {emotion}
+            </StyledFilterButtons>
+          </StyledFilterList>
+        ))}
+      </ul>
+
+      {filteredObjects.length === 0 ? (
         <StyledMessage>
           At the moment there are no emotions in the list. Please add an
           emotion.
         </StyledMessage>
       ) : (
-        objects.map((object) => (
-          <StyledLink key={object.id} href={`emotion/${object.id}`}>
-            <li>
-              <EmotionCard object={object} />
-            </li>
-          </StyledLink>
-        ))
+        <ul>
+          {filteredObjects.map((object) => (
+            <StyledCardList key={object.id}>
+              <StyledLink StyledLink href={`emotion/${object.id}`}>
+                <EmotionCard object={object} />
+              </StyledLink>
+            </StyledCardList>
+          ))}
```

Comment:
Make sure to update `filteredObjects` to the new name here as well, based on the suggestion above,  Also, consider renaming `object` in this map function to something more descriptive, `emotionItem`, for better readability. 

💡 Note: When renaming properties like `object` to `emotionItem`, make sure to update all references  within the component to avoid errors. 

```suggestion
          {filteredEmotions.map((emotionItem) => (
            <StyledCardList key={emotionItem.id}>
              <StyledLink StyledLink href={`emotion/${emotionItem.id}`}>
                <EmotionCard emotionItem={emotionItem} />
              </StyledLink>
            </StyledCardList>
          ))}
```